x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/StringUtilsTests.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.expression.function.scalar.string;
+
+import org.elasticsearch.test.ESTestCase;
+
+import static org.elasticsearch.xpack.eql.expression.function.scalar.string.StringUtils.substringSlice;
+
+public class StringUtilsTests extends ESTestCase {
+
+    public void testSubstringSlicePositive() {
+        String str = randomAlphaOfLength(10);
+        assertEquals(str.substring(1, 7), substringSlice(str, 1, 7));
+    }
+
+    public void testSubstringSliceNegative() {
+        String str = randomAlphaOfLength(10);
+        assertEquals(str.substring(5, 9), substringSlice(str, -5, -1));
+    }
+
+    public void testSubstringSliceNegativeOverLength() {
```

Comment:
How about both negative and both higher than length?
+ both positive and both higher than end

Reply:
That's what the test does - `-15` and `-11` are both negative and higher than length.
Am I missing something?

Reply:
Sorry my bad about the negative.